### PR TITLE
MULE-16669: Mule throws error Illegal group reference Exception when …

### PR DESCRIPTION
**Pull request by JuLuzzi**
…trying to pass $ into URI Parameter in HTTP request

### CODE REVIEW

File: src/main/java/org/mule/extension/http/api/request/builder/HttpRequesterRequestBuilder.java [hunk 1]
```diff
@@ -115,7 +116,7 @@ public String replaceUriParams(String path) {
         throw new NullPointerException(format("Expression {%s} evaluated to null.", uriParamName));
       }
 
-      path = path.replaceAll("\\{" + uriParamName + "\\}", uriParamValue);
+      path = path.replaceAll(format("\\{%s}", uriParamName), quoteReplacement(uriParamValue));
```
Comment (afelisatti) [inline src/main/java/org/mule/extension/http/api/request/builder/HttpRequesterRequestBuilder.java]: Why is the format necessary?

File: src/test/java/org/mule/test/http/functional/requester/HttpRequestUriParamsTestCase.java [hunk 2]
```diff
@@ -63,4 +63,19 @@ public void sendsUriParamsIfNull() throws Exception {
     flowRunner("uriParamNull").run();
   }
 
+  @Test
+  public void uriParamsContainsReservedUriCharacter() throws Exception {
+    flowRunner("reservedUriCharacter").withPayload(AbstractMuleContextTestCase.TEST_MESSAGE)
```
Comment (afelisatti) [inline src/test/java/org/mule/test/http/functional/requester/HttpRequestUriParamsTestCase.java]: static import

File: src/main/java/org/mule/extension/http/api/request/builder/HttpRequesterRequestBuilder.java [hunk 3]
```diff
@@ -115,7 +116,7 @@ public String replaceUriParams(String path) {
         throw new NullPointerException(format("Expression {%s} evaluated to null.", uriParamName));
       }
 
-      path = path.replaceAll("\\{" + uriParamName + "\\}", uriParamValue);
+      path = path.replaceAll("\\{" + uriParamName + "}", quoteReplacement(uriParamValue));
```
Comment (afelisatti) [inline src/main/java/org/mule/extension/http/api/request/builder/HttpRequesterRequestBuilder.java]: I think you deleted extra `\\`

Comment (JuLuzzi) [reply src/main/java/org/mule/extension/http/api/request/builder/HttpRequesterRequestBuilder.java]: Only the opening brace needs to be scaped

**Pull request merged**